### PR TITLE
Explain database transactions.

### LINKED ISSUE #431: Documet database transactions management.
Handle:
* single step
* whole substory

### PR DESCRIPTION
**Pull request by proofit404**
Fix #431

### CODE REVIEW

**Comment by proofit404** [comment]
@supadrupa, please review

**Comment by supadrupa** [comment]
I want a more detailed example for django

**Comment by proofit404** [comment]
Django related documentation will be written as part of the #435 

Thanks!

File: docs/transactions.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+>>> persistance = Persistance()
+>>> purchase = Purchase(lock_item, charge_money, send_notification)
+>>> transactional = Transactional(persistance.start_transaction, purchase.make, persistance.end_transaction)
+>>> transactional.do.run(user_id=2)
+BEGIN TRANSACTION;
+UPDATE 'items';
+UPDATE 'balance';
+Failure()
+
+>>> persistance.finalize()
```
Comment (proofit404) [inline docs/transactions.md]:
```python
try:
    transactional.do.run(user_id=1)
finally:
    persistance.finalize()
```

**Comment by proofit404** [comment]
Superseded by #663

**Pull request closed without merging**